teams_devices.go

```diff
@@ -0,0 +1,105 @@
+package cloudflare
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+type TeamsDevicesList struct {
+	Response
+	Result []TeamsDeviceListItem `json:"result"`
+}
+
+type TeamsDeviceDetail struct {
+	Response
+	Result TeamsDeviceListItem `json:"result"`
+}
+
+type TeamsDeviceListItem struct {
+	User         UserItem `json:"user,omitempty"`
+	ID           string   `json:"id,omitempty"`
+	Key          string   `json:"key,omitempty"`
+	DeviceType   string   `json:"device_type,omitempty"`
+	Name         string   `json:"name,omitempty"`
+	Model        string   `json:"model,omitempty"`
+	Manufacturer string   `json:"manufacturer,omitempty"`
+	Deleted      bool     `json:"deleted,omitempty"`
+	Version      string   `json:"version,omitempty"`
+	SerialNumber string   `json:"serial_number,omitempty"`
+	OSVersion    string   `json:"os_version,omitempty"`
+	MacAddress   string   `json:"mac_address,omitempty"`
+	IP           string   `json:"ip,omitempty"`
+	Created      string   `json:"created,omitempty"`
+	Updated      string   `json:"updated,omitempty"`
+	LastSeen     string   `json:"last_seen,omitempty"`
+	RevokedAt    string   `json:"revoked_at,omitempty"`
+}
+
+type UserItem struct {
+	ID    string `json:"id,omitempty"`
+	Name  string `json:"name,omitempty"`
+	Email string `json:"email,omitempty"`
+}
+
+// ListTeamsDevice returns all devices for a given account.
+//
+// API reference : https://api.cloudflare.com/#devices-list-devices
+func (api *API) ListTeamsDevices(ctx context.Context, accountID string) ([]TeamsDeviceListItem, error) {
+	uri := fmt.Sprintf("/%s/%s/devices", AccountRouteRoot, accountID)
+
+	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
+	if err != nil {
+		return []TeamsDeviceListItem{}, err
+	}
+
+	var response TeamsDevicesList
+	err = json.Unmarshal(res, &response)
+	if err != nil {
+		return []TeamsDeviceListItem{}, errors.Wrap(err, errUnmarshalError)
+	}
+
+	return response.Result, nil
+}
+
+// RevokeTeamsDevice revokes device with given identifiers
+//
+// API reference : https://api.cloudflare.com/#devices-revoke-devices
+func (api *API) RevokeTeamsDevices(ctx context.Context, accountID string, deviceIds []string) (Response, error) {
+	uri := fmt.Sprintf("/%s/%s/devices/revoke", AccountRouteRoot, accountID)
+
+	res, err := api.makeRequestContextWithHeaders(ctx, http.MethodPost, uri, deviceIds, http.Header{"Content-Type": []string{"application/json"}})
```

Comment:
you shouldn't need the additional HTTP header here as the library appends it by default.

https://github.com/cloudflare/cloudflare-go/blob/196b1d80bb771df37612cba0a978b23a49a163a1/cloudflare.go#L317-L319

```suggestion
	res, err := api.makeRequestContextWithHeaders(ctx, http.MethodPost, uri, deviceIds)
```